src/Model/Profile.php

```diff
@@ -84,6 +86,69 @@ public static function getListByUser(int $uid, array $fields = [])
 		return DBA::selectToArray('profile', $fields, ['uid' => $uid]);
 	}
 
+	/**
+	 * Update a profile entry and distribute the changes if needed
+	 *
+	 * @param array $fields
+	 * @param integer $uid
+	 * @return boolean
+	 */
+	public static function update(array $fields, int $uid): bool
+	{
+		$old_owner = User::getOwnerDataById($uid);
+		if (empty($old_owner)) {
+			return false;
+		}
+
+		if (!DBA::update('profile', $fields, ['uid' => $uid])) {
+			return false;
+		}
+
+		$update = Contact::updateSelfFromUserID($uid);
+
+		$owner = User::getOwnerDataById($uid);
+		if (empty($owner)) {
+			return false;
+		}
+
+		if ($old_owner['name'] != $owner['name']) {
+			User::update(['username' => $owner['name']], $uid);
+		}
+
+		$profile_fields = ['postal-code', 'dob', 'prv_keywords', 'homepage'];
+		foreach ($profile_fields as $field) {
+			if ($old_owner[$field] != $owner[$field]) {
+				$update = true;
+			}
+		}
+
+		if ($update) {
+			self::publishUpdate($uid);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Publish a changed profile
+	 * @param int $uid 
+	 */
+	public static function publishUpdate(int $uid)
+	{
+		$owner = User::getOwnerDataById($uid);
+		if (empty($owner)) {
+			return;
+		}
+		if ($owner['net-publish']) {
```

Comment:
What happens when someone deselects this option from their profile after it had been selected? Shouldn't we ask the directory to remove this profile? In this state the profile may stay in the global directory as it last was when it was still searchable.

Unless the directory does this kind of housekeeping itself the next time it probes the profile URL, but in this case it would also be better to trigger a probe immediately to hide the profile from the directory.

Reply:
You know the directory system better than me. Does it already do so?

Reply:
I can confirm we remove the profile from the directory when we encounter the `hide` parameter even if the profile was available before. But it only happens when the profile is polled.